components/supervisor/cmd/init.go

```diff
@@ -81,61 +65,56 @@ var initCmd = &cobra.Command{
 		case <-supervisorDone:
 			// supervisor has ended - we're all done here
 			return
-		case s := <-sigSupervisor:
+		case <-sigInput:
 			// we received a terminating signal - pass on to supervisor and wait for it to finish
-			_ = runCommand.Process.Signal(s)
-			<-supervisorDone
+			timer := time.After(cfg.GetTerminationGracePeriod())
+			terminationDone := make(chan int)
```

Comment:
```suggestion
			terminationDone := make(chan struct{})
```
is more idiomatic and uses fewer bytes